## 1. Analisis del dataset

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [11]:
# Configuracion para que los graficos se vean limpios
sns.set_theme(style="whitegrid")
%matplotlib inline

In [12]:
# Se carga el dataset una vez limpio
df = pd.read_csv('pokemon.csv')

In [13]:
# Comprobamos cuantas filas y columnas tiene el dataset
print(f"El dataset tiene: {df.shape[0]} filas y {df.shape[1]} columnas.")

El dataset tiene: 1025 filas y 15 columnas.


In [14]:
# Ahora vemos las primeras 5 filas
df.head()

,pokedex_number,name,type_1,hp,attack,defense,sp_attack,sp_defense,speed,base_stat_total,generation,is_legendary,is_mythical,habitat,capture_rate
0,1,Bulbasaur,Grass,45,49,49,65,65,45,318,1,False,False,grassland,45
1,2,Ivysaur,Grass,60,62,63,80,80,60,405,1,False,False,grassland,45
2,3,Venusaur,Grass,80,82,83,100,100,80,525,1,False,False,grassland,45
3,4,Charmander,Fire,39,52,43,60,50,65,309,1,False,False,mountain,45
4,5,Charmeleon,Fire,58,64,58,80,65,80,405,1,False,False,mountain,45


In [15]:
# Por ultimo comprobaremos los tipos de datos de cada columna
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1025 entries, 0 to 1024
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   pokedex_number   1025 non-null   int64
 1   name             1025 non-null   str  
 2   type_1           1025 non-null   str  
 3   hp               1025 non-null   int64
 4   attack           1025 non-null   int64
 5   defense          1025 non-null   int64
 6   sp_attack        1025 non-null   int64
 7   sp_defense       1025 non-null   int64
 8   speed            1025 non-null   int64
 9   base_stat_total  1025 non-null   int64
 10  generation       1025 non-null   int64
 11  is_legendary     1025 non-null   bool 
 12  is_mythical      1025 non-null   bool 
 13  habitat          386 non-null    str  
 14  capture_rate     1025 non-null   int64
dtypes: bool(2), int64(10), str(3)
memory usage: 106.2 KB


## 2. Limpieza y gestion de valores

El fichero csv original disponia de 31 columnas y 1350 filas, a partir de ahi se ha aplicado el siguiente criterio de limpieza:
- Reduccion a 15 columnas relevantes: Se detectaron datos irrelevantes al analisis los cuales se borraron para centrarnos en las metricas necesarias .
- Reduccion a 1025 filas: Oficialmente hay 1025 pokemon, originalmente en el csv habian 1350 filas siendo a partir de la 1026 variantes de pokemon, como las mega evoluciones o formas de estos lo que generaba duplicados ya que estas no disponen de un numero de pokedex oficial.
- Remedio para la columna 'habitat': Se comprobara a continuacion si hay valores nulos.

#### 2.1 Comprobacion y razonamiento de cuantos valores nulos hay en cada columna

In [16]:
nulos = df.isnull().sum()
nulos[nulos > 0]

habitat    639
dtype: int64

De los 1025 Pokemon, 386 tienen un habitat asignado mientras que los 639 restantes tienen ese valor vacio. Esto se debe a que la asignacion por habitat solo se utilizo oficialmente para las generaciones 1, 2, y 3, por esa razon es que a partir de la generacion 4 los pokemon no tienen un 'habitat'.

## 3. Estadisticas descriptivas globales

Calculamos las principales estadisticas (medias, minimo, maximo, etc...) para las caracteristicas numericas de combate y el ratio de captura de cada pokemon 

In [17]:
# Seleccionamos las columnas numericas (estadisticas) y de ratio de captura
cols_stats = ['hp', 'attack', 'defense', 'sp_attack', 'sp_defense', 'speed', 'base_stat_total', 'capture_rate']

In [18]:
# Calculamos las estadisticas y las redondeamos a 2 decimales
df[cols_stats].describe().round(2)

,hp,attack,defense,sp_attack,sp_defense,speed,base_stat_total,capture_rate
count,1025.00,1025.00,1025.00,1025.00,1025.00,1025.00,1025.00,1025.00
mean,70.18,77.52,72.51,70.08,70.21,67.19,427.69,95.45
std,26.63,29.78,29.29,29.66,26.64,28.72,112.77,75.99
min,1.00,5.00,5.00,10.00,20.00,5.00,175.00,3.00
25%,50.00,55.00,50.00,47.00,50.00,45.00,325.00,45.00
50%,68.00,75.00,70.00,65.00,67.00,65.00,450.00,60.00
75%,85.00,100.00,90.00,90.00,86.00,88.00,508.00,140.00
max,255.00,181.00,230.00,173.00,230.00,200.00,720.00,255.00


## 4. Conclusiones del Analisis inicial

- Dataset validado: Se disponen de 1025 registros (cada uno siendo una especie de pokemon unica) y 15 columnas clave para la busqueda.
- Calidad de los datos: Todas las metricas numericas de combate y/o atributos han sido completados al 100% exceptuando la columna de habilidad, la cual se gestionara filtrandola a la hora de responder las preguntas correspondientes.
- Siguiente paso: A partir de este analisis inicial y que los datos esten validados, el siguiente paso es proceder a responder las 5 preguntas de investigacion en sus notebooks correspondientes.